File: python/day01/day01.py
from itertools import groupby


def parse_carried_calories(lines: list[str]) -> list[int]:
    return [
        sum(int(item) for item in group)
        for filter, group in groupby(lines, lambda line: line != '')
        if filter
    ]


with open('day01/input', encoding='utf-8') as f:
    lines = [line.strip() for line in f]

calories_carried = parse_carried_calories(lines)
calories_carried.sort()
print(f'Part 1: {calories_carried[-1]}')
print(f'Part 2: {sum(calories_carried[-3:])}')

File: typescript/src/day01/day01.ts
import fs from "fs";

const caloriesCarried = fs
  .readFileSync("src/day01/input")
  .toString("utf-8")
  .split("\n\n")
  .map((group) =>
    group
      .split("\n")
      .map((item) => parseInt(item))
      .filter((item) => !Number.isNaN(item))
      .reduce((a, b) => a + b, 0)
  )
  .sort();

const topOne = caloriesCarried[caloriesCarried.length - 1];
const topThree = caloriesCarried
  .slice(caloriesCarried.length - 3, caloriesCarried.length)
  .reduce((a, b) => a + b, 0);

console.log("Part 1: " + topOne);
console.log("Part 2: " + topThree);
